module/os/linux/spl/spl-kmem.c

```diff
@@ -245,7 +245,21 @@ spl_kmem_alloc_impl(size_t size, int flags, int node)
 				return (NULL);
 			}
 		} else {
-			if (flags & KM_VMEM) {
+			/*
+			 * We use kmalloc when doing kmem_alloc(KM_NOSLEEP),
+			 * because kvmalloc/vmalloc may sleep.  We also use
+			 * kmalloc on systems with limited kernel VA space (e.g.
+			 * 32-bit), which have HIGHMEM.  Otherwise we use
+			 * kvmalloc, which tries to get contiguous physical
+			 * memory (fast, like kmalloc) and falls back on using
+			 * virtual memory to stitch together pages (slow, like
+			 * vmalloc).
+			 */
+#ifdef CONFIG_HIGHMEM
+			if (flags & KM_VMEM)) {
```

Comment:
```suggestion
			if (flags & KM_VMEM) {
```

```
/var/lib/buildbot/slaves/buildslave2/Ubuntu_16_04_i386__BUILD_/build/zfs/module/spl/../os/linux/spl/spl-kmem.c: In function ‘spl_kmem_alloc_impl’:
/var/lib/buildbot/slaves/buildslave2/Ubuntu_16_04_i386__BUILD_/build/zfs/module/spl/../os/linux/spl/spl-kmem.c:259:24: error: expected statement before ‘)’ token
    if (flags & KM_VMEM)) {
                        ^
```

Reply:
hm, yes it would help if I could copy/paste correctly.